### PR TITLE
MOTECH-2694: Added element id attribute naming convention

### PR DESCRIPTION
**Pull request by jhopen**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: docs/source/development/coding_conventions.rst [hunk 1]
```diff
@@ -231,6 +231,56 @@ corresponds to complexity of the body).
 
 |x| **Do not** declare imports not used within the file.
 
+Element ID
+----------
```
Comment (nickdotreid) [inline docs/source/development/coding_conventions.rst]: @jhopen Can we add a sentence or two to reflect why we are adding element ids? I feel that showing the reason and not just the rule will help guide devs... and of course we didn't write that yet

Here is the start of something, please let me know if you think it makes sense:

> To allow for easier integration testing of the MOTECH-UI, we are using a naming convention for the element IDs of navigational and action based items in the UI. The element IDs should follow the format below, unless there is a clear reason the format shouldn't be followed.

Comment (jhopen) [inline docs/source/development/coding_conventions.rst]: @nickdotreid I am not sure about that, there is Introduction with Goals and Guideline Presentation at the beginning of document that explains why and how to use it so adding more info about that later in doc feels a bit redundant. I will ask Sebastian what he thinks about it.

Comment (jhopen) [inline docs/source/development/coding_conventions.rst]: Sure, I will add some short info why to use it. What You wrote sounds good.

Comment (sebbrudzinski) [inline docs/source/development/coding_conventions.rst]: I think it makes sense to add a **short** motivation for the id naming convention. This is a little different than what the introduction of this document covers, so I've got no problem with that.

**Comment by sebbrudzinski** [comment]
@nickdotreid The fix has been uploaded. Does everything else look good to you?

**Comment by nickdotreid** [comment]
@sebbrudzinski & @jhopen 

Looks great! Squash the commits and merge it!

File: docs/source/development/coding_conventions.rst [hunk 2]
```diff
@@ -273,7 +273,16 @@ The entity id refers to a repeating id which is used in lists of links.
 
 |v| **Consider** using appropriate non-numeric id, in preference to a numeric id.
 
-4. Action
+4. Error (optional)
+If element is used for error messages use ".errors." before action.
+
+.. code-block:: html
+	<lu id="email.send.subject-errors">
```
Comment (sebbrudzinski) [inline docs/source/development/coding_conventions.rst]: the correct tag is "ul"

File: docs/source/development/coding_conventions.rst [hunk 3]
```diff
@@ -273,7 +273,16 @@ The entity id refers to a repeating id which is used in lists of links.
...
+		<li id="email.send.subject.errors.lenght">message</li>
```
Comment (sebbrudzinski) [inline docs/source/development/coding_conventions.rst]: length

**Comment by sebbrudzinski** [comment]
test this please

**Pull request merged**